qutip/quantum_systems/jaynes_cummings.py

```diff
@@ -0,0 +1,170 @@
+    operators['sigma_minus'] = tensor(qeye(n_cavity), destroy(2))
+    operators['sigma_plus'] = operators['sigma_minus'].dag()
+    operators['sigma_z'] = tensor(qeye(n_cavity), sigmaz())
+    operators['sigma_x'] = tensor(qeye(n_cavity), sigmax())
+    operators['sigma_y'] = tensor(qeye(n_cavity), sigmay())
+
+    # Build Hamiltonian
+    # Free cavity evolution: ω_c a†a
+    H_cavity = omega_c * operators['n_c']
+
+    # Free atomic evolution: (ω_a/2)σ_z
+    H_atom = omega_a * operators['sigma_plus'] * operators['sigma_minus']
+
+    # Interaction term
+    if rotating_wave:
+        # Rotating wave approximation: g(a†σ_- + aσ_+)
+        H_interaction = g * (operators['a_dag'] * operators['sigma_minus'] +
+                             operators['a'] * operators['sigma_plus'])
+    else:
+        # Full interaction (Rabi model): g(a† + a)(σ_+ + σ_-)
+        H_interaction = g * (operators['a_dag'] + operators['a']) * \
+            (operators['sigma_plus'] + operators['sigma_minus'])
+
+    # Total Hamiltonian
+    hamiltonian = H_cavity + H_atom + H_interaction
+
+    # Build collapse operators for dissipation
+    c_ops = []
+
+    # Cavity decay with thermal effects 
+    if isinstance(cavity_decay, Coefficient):
+        # cavity_decay is coefficient, thermal_photons is float
+        def cavity_relax_func(t, args):
+            kappa = cavity_decay(t, args)
+            return np.sqrt(kappa * (1 + thermal_photons))
+        
+        sqrt_cavity_relax = coefficient(cavity_relax_func, args={})
+        c_ops.append(sqrt_cavity_relax * operators['a'])
+    elif cavity_decay > 0.0:  
+        cavity_relax_rate = cavity_decay * (1 + thermal_photons)
+        c_ops.append(np.sqrt(cavity_relax_rate) * operators['a'])
+
+    # Cavity excitation (thermal): sqrt(kappa * n_th) * a_dag  
+    # Cavity excitation (thermal): sqrt(kappa * n_th) * a_dag  
+    if thermal_photons > 0.0:
+        if isinstance(cavity_decay, Coefficient):
+            # cavity_decay is coefficient, thermal_photons is float
+            def cavity_excite_func(t, args):
+                kappa = cavity_decay(t, args)
+                return np.sqrt(kappa * thermal_photons)
+            
+            sqrt_cavity_excite = coefficient(cavity_excite_func, args={})
+            c_ops.append(sqrt_cavity_excite * operators['a_dag'])
+        elif cavity_decay > 0.0: 
+            cavity_excite_rate = cavity_decay * thermal_photons
+            c_ops.append(np.sqrt(cavity_excite_rate) * operators['a_dag'])
+
+    # Atomic spontaneous emission: sqrt(gamma) * sigma_minus
+    # Check if it's a Coefficient
+    if isinstance(atomic_decay, Coefficient):
+        sqrt_atomic_decay = _create_sqrt_coefficient(atomic_decay)
+        c_ops.append(sqrt_atomic_decay * operators['sigma_minus'])
+    elif atomic_decay > 0.0:  
+        sqrt_atomic_decay = _create_sqrt_coefficient(atomic_decay)
+        c_ops.append(sqrt_atomic_decay * operators['sigma_minus'])
```

Comment:
In the cavity decay case, it was not the same code for the Coefficient and scalar cases. Here it is the same...

```suggestion
    if isinstance(atomic_decay, Coefficient) or atomic_decay > 0.0:
        sqrt_atomic_decay = _create_sqrt_coefficient(atomic_decay)
        c_ops.append(sqrt_atomic_decay * operators['sigma_minus'])
```